pkg/vsphere/compute/placement/random.go

```diff
@@ -32,28 +32,31 @@ func NewRandomHostPolicy() *RandomHostPolicy {
 }
 
 // CheckHost always returns false in a RandomHostPolicy.
-func (p *RandomHostPolicy) CheckHost(op trace.Operation, vm *vm.VirtualMachine) bool {
+func (p *RandomHostPolicy) CheckHost(op trace.Operation, sess *session.Session) bool {
 	return false
 }
 
 // RecommendHost recommends a random host on which to place a newly created VM.
-func (p *RandomHostPolicy) RecommendHost(op trace.Operation, vm *vm.VirtualMachine) (*object.HostSystem, error) {
-	r, err := vm.ResourcePool(op)
-	if err != nil {
-		return nil, err
+func (p *RandomHostPolicy) RecommendHost(op trace.Operation, sess *session.Session, hosts []*object.HostSystem) ([]*object.HostSystem, error) {
```

Comment:
Once this branch is rebased to `feature/robo` we should update the calls to the placement lib during VCH creation.

https://github.com/vmware/vic/blob/feature/robo/lib/install/management/create.go#L125-L128
https://github.com/vmware/vic/blob/feature/robo/lib/install/management/create.go#L135-L149